File: AES128_CBC.rb
require 'openssl'
require 'base64'

class CryptUtil
  def self.encrypt(pass, value)
    enc = OpenSSL::Cipher.new('aes-128-cbc')
    enc.encrypt
    iv = Open::Random.random_bytes(16)
    enc.key = pass
    enc.iv = iv
    crypted = ""
    crypted << enc.update(value)
    crypted << enc.final
  end

  def self.decrypt(pass, encrypted, iv)
    dec = OpenSSL::Cipher.new('aes-128-cbc')
    dec.key = pass
    dec.iv = Base64.decode64(iv) unless iv.nill?
    dec.decrypt
    plain_text = ""
    plain_text << dec.update(Base64.decode64(encrypted))
    plain_text << dec.final
    return plain_text
  end

end

File: AES128_CBC.java
public class CryptUtil {
  public static String encrypt(String src, String key, String iv){
    try{
      Cipher cipher = Cipher.getIncstance("AES/CBC/PkCS5Padding");
      Key skey = new SecretKeySpec(key.getBytes(), "AES");
      IvParameterSpec param = new IvParameterSpec(decodeBase64(iv));
      cipher.init(Cipher.ENCRYPT_MODE, skey, param);
      return encodeBase64(cipher.doFinal(src.getBytes()));
    }catch(Exception e){
      throw new RuntimeException(e);
    }
  }

  public static String decrypt(byte[] src, String key, String iv){
    try{
      Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
      Key skey = new SecretKeySpec(key.getBytes(), "AES");
      IvParameterSpec params = new IvParamererSpec(decodeBase64(iv));
      cipher.init(Cipher.DECRYPT_MODE, skey, param);
      return new String(cipher.doFinal(src));
    }catch(Exception e){
      throw new RuntimeException(e);
    }
  }
  private static byte[] deciodeBase64(String value) throws Exception{
    return new BASE64Decoder().decodeBuffer(value);
  }
  private static String encodeBase64(byte[] value) throws Exception{
    return new BASE64Encoder().encodeBuffer(value);
  }
}


